File: backend/app/schemas/ingestion.py
from typing import Any, Dict, List, Optional
from datetime import datetime
from pydantic import BaseModel, HttpUrl

class IngestionUrlRequest(BaseModel):
    project_id: str
    url: str
    title: Optional[str] = None

class IngestionTextRequest(BaseModel):
    project_id: str
    title: str
    text: str

class IngestionResponse(BaseModel):
    ingestion_id: str
    source_document_id: str
    project_id: str
    status: str
    title: str
    source_type: str
    created_at: datetime

    class Config:
        from_attributes = True

class NormalizedDocumentSchema(BaseModel):
    document_id: str
    source_type: str
    title: str
    text: str
    language: Optional[str] = "English"
    metadata: Dict[str, Any] = {}
    pages: List[Dict[str, Any]] = []
    media: List[Dict[str, Any]] = []
    entities: List[str] = []
    timestamps: List[Dict[str, Any]] = []

File: frontend/src/types/ingestion.ts
export interface NormalizedDocument {
  document_id: string;
  source_type: 'pdf' | 'docx' | 'txt' | 'image' | 'audio' | 'video' | 'url' | 'text';
  title: string;
  text: string;
  language?: string;
  metadata?: Record<string, any>;
  pages?: Array<{ page_number: number; text: string }>;
  media?: Array<{ type: string; path: string }>;
  entities?: string[];
  timestamps?: Array<{ start: number; end: number; text: string }>;
}

export interface IngestionResponse {
  ingestion_id: string;
  source_document_id: string;
  project_id: string;
  status: 'uploaded' | 'processing' | 'completed' | 'failed';
  title: string;
  source_type: string;
  created_at: string;
}
